src/components/limited-restoration-panel/LimitedRestorationPanel.vue

```diff
@@ -0,0 +1,104 @@
+<template id="limited-restoration-panel">
+  <v-radio-group class="mt-0 pt-0 ml-8" v-model="selectMonths" mandatory>
+    <v-radio id="twentyFour-radio" name="selectMonths" label="Two Years" :value=24 />
+    <v-radio id="eighteen-radio" name="selectMonths" label="18 months" :value=18 />
+    <v-radio id="twelve-radio" name="selectMonths" label="12 months" :value=12 />
+    <v-radio id="six-radio" name="selectMonths" label="6 months" :value=6 />
+    <v-row class="ml-0">
+      <v-radio id="custom-months" name="selectMonths" label="Month(s): " value="customMonths">
+      </v-radio>
+      <v-text-field class="shrink mx-4" type="number" min=1 max=24 :rules="monthRules" v-model="numberOfMonths"
+        :disabled="!customMonths" @change="onMonthsChanged" filled />
+    </v-row>
+  </v-radio-group>
+</template>
+<script lang="ts">
+
+import { Component, Emit, Mixins, Prop, Watch } from 'vue-property-decorator'
+import { DateMixin } from '@/mixins' // NB: local mixin (StoryBook can't find it otherwise)
+
+@Component({})
+export default class LimitedRestorationPanel extends Mixins(DateMixin) {
+  @Prop({ default: '' }) readonly currentDate!: string
+
+  @Prop({ default: '' }) readonly expiryDate!: string
+
+  // Local properties
+  private customMonths = false
+  private selectMonths = ''
+  private numberOfMonths = 1
+  private monthRules: ((data: string) => void)[] = []
+
+  /**
+   * Called when the component is mounted.
+   * Auto select the previous months input of user (if any).
+   * If there is no previous input, auto select 24 months.
+   */
+  mounted (): void {
+    this.setMonthRules()
+    if (this.expiryDate !== null && this.expiryDate !== undefined) { // Unit tests fail without the first condition.
```

Comment:
This chec kshould not be needed. You have provided default values for both of these, so they should never be null or undefined here.

However, your default values are not computable, and it's possible the caller passed in non-computable values, so you might want to check that.